src/experimental/tools/videoThumbnailLoader/thumbnail_loader.ts

```diff
@@ -0,0 +1,245 @@
+                        this._videoElement).pipe(
+        mergeMap((videoSourceBuffer) => {
+          const removeBuffers$: Observable<unknown> =
+            removeBufferForTime$(this._videoElement, videoSourceBuffer, time);
+          return removeBuffers$.pipe(
+            mergeMap(() => {
+              log.debug("VTL: Removed buffer before appending segments.", time);
+
+              return segmentLoader({
+                manifest: contentInfos.manifest,
+                period: contentInfos.manifest.periods[0],
+                adaptation: contentInfos.adaptation,
+                representation: contentInfos.representation,
+                segment,
+              }).pipe(
+                filter((evt): evt is { type: "data";
+                                       value: { responseData: Uint8Array }; } =>
+                  evt.type === "data"),
+                mergeMap((evt) => {
+                  const inventoryInfos = { manifest: contentInfos.manifest,
+                                           period: contentInfos.period,
+                                           adaptation: contentInfos.adaptation,
+                                           representation: contentInfos.representation,
+                                           segment };
+                  return videoSourceBuffer
+                    .pushChunk({ data: { chunk: evt.value.responseData,
+                                         timestampOffset: 0,
+                                         appendWindow: [undefined, undefined],
+                                         initSegment: null,
+                                         codec: contentInfos
+                                           .representation.getMimeTypeString() },
+                                 inventoryInfos })
+                      .pipe(map(() => {
+                        log.debug("VTL: Appended segment.", evt.value.responseData);
+                        this._videoElement.currentTime = time;
+                        if (this._videoElement.buffered.length === 0) {
+                          throw new VideoThumbnailLoaderError("NOT_BUFFERED",
+                                                              "No buffered data after loading.");
+                        } else {
+                          return time;
+                        }
+                      })
+                    );
+                })
+              );
+            })
+          );
+        }),
+        catchError((err: Error | { message?: string; toString(): string }) => {
+          const newError =
+            new VideoThumbnailLoaderError("LOADING_ERROR",
+                                          (err.message ?? err.toString()));
+          throw newError;
+        })
+      ),
+      abortError$
+    ).pipe(take(1)).toPromise(PPromise)
+      .then((res) => { this._currentJob = undefined; return res; })
+      .catch((err) => { this._currentJob = undefined; throw err; });
+
+    this._currentJob = {
```

Comment:
We should complete the previous killJob$ somewhere to make sure there is no leak

Reply:
Done